R/admiraldev_environment.R

```diff
@@ -0,0 +1,23 @@
+#' Environment Objects
+#'
+#' @details
+#' Once in a while, we may encounter "locked binding for 'xxx'." errors
+#' during the development process while building out functions. This may arise because
+#' we want to create dynamic data/objects based on user-inputs that need modification
+#' at points in time after the package has been loaded. To manage such data or objects,
+#' R has a data structure known as an 'environment'. These environment objects are created
+#' at build time, but can be populated with values after the package has been loaded and
+#' update those values over the course of an R session. See Chapter 8.5 of the R Packages textbook
+#' or Chapter 7 of the Advanced R textbook for more details.
```

Comment:
I would say something like "see relevant sections on environments in...." and not list the chapters as those books change every 6 months.